modules/exploits/multi/misc/weblogic_deserialize_badattrval.rb

```diff
@@ -62,7 +63,10 @@ def initialize(info = {})
       )
     )
 
-    register_options([ Opt::RPORT(7001) ])
+    register_options([
+        Opt::RPORT(7001),
+        OptString.new('PROTOCOL', [true, 'T3 vs T3S protocol', 'T3'])
```

Comment:
1 - should this be an advanced option? 2 - should it be boolean?

Reply:
> 1 - should this be an advanced option? 2 - should it be boolean?

An `OptEnum` would be more suitable.

```suggestion
        OptEnum.new('PROTOCOL', [true, 'T3 vs T3S protocol.', 'T3', %w[T3 T3S]]),
```

However, as T3S appears to simply be T3 over SSL/TLS, leveraging the existing TCP `SSL` option would be more suitable.

Reply:
Was unsure on this because it might break existing uses for the pre-existing SSL option. (all I can think though that may be harmed would be "standard" T3 tunnelled through some other means such stunnel).

Reply:
Only alternative that comes to mind and wouldn't break that use case I guess would be to have a "Force T3 Handshake" advanced option, so it _defaults_ to T3S when SSL enabled, but can be gracefully reverted to T3 over SSL? This was it may still break existing working shells, but can be reverted by users.

I don't see the other way around being much of an issue, i.e T3S over unencrypted. I imagine that may exist but rare enough to not be worth considering?